### ISSUE TITLE
metric used to calculate all the dissimilarity matrices

### ISSUE THREAD

**Issue by Warhorze** [zcang/SpaOTsc#1]
Hi,

I am trying to reproduce the experiments but I can't find which distance metric is used to construct the dissimilarity matrices. Would you be so kind to provide the code or tell me which metric is used? 

Thank you in advance!

**Comment by zcang** [comment]
Hi,

Thanks for asking.

We found Matthew's correlation coefficient or Spearman correlation coefficient effective when there is a relatively small group of carefully selected genes, e.g. the Drosophila embryo example with 84 genes in spatial reference. These correlation coefficients were translated to dissimilarity by $\exp(-c)$. When the spatial dataset is high-throughput such as Slideseq and you would like to use all common genes measured in the two datasets, a current good practice is to use [Scanorama](http://cb.csail.mit.edu/cb/scanorama/) to map the datasets to a common low dimensional space and use the Cosine distance or simply Euclidean distance.

**Comment by Warhorze** [comment]
Thank you for your quick response, I first tried it with the  `matthews_corrcoef`from `sklearn.metric`. that didn't give me the same result. However, when I converted the R code from the `DistMap` repo to python it worked. 

Here is the python code if anyone needs it.

`
def MCC_metric(w1,w2):
    mt = w1 - w2
    FP = np.sum(mt == 2)
    TP = np.sum(mt == 1)
    TN = np.sum(mt == 0)
    FN = np.sum(mt == -1)
    MCC  = (TP*TN - FP*FN)/np.sqrt((TP+FP)*(TP+FN)*(TN+FP)*(TN+FN))
    if np.isnan(MCC):
        MCC = 0
    return(MCC)

M = cdist(df_is_bin, df_sc_bin, metric = matthews_corrcoef).T
`
